### PR TITLE
update template mixin and opAssign spec to reflect what compiler actually does

### PR DESCRIPTION
**Pull request by WalterBright**
Based on suggestions from https://issues.dlang.org/show_bug.cgi?id=15957

### CODE REVIEW

File: spec/struct.dd [hunk 1]
```diff
@@ -521,8 +521,20 @@ $(SECTION3 $(LEGACY_LNAME2 AssignOverload, assign-overload, Identity Assignment
...
+	$(LI it has a $(LINK2 #struct-destructor, destructor))
+	$(LI it has a $(LINK2 #struct-postblit, postblit))
+	$(LI it has a field with an identity assignment overload)
+	)
+
+	$(P If an identity assignment overload is required and does not
+	exist, an identity assignment overload function of the type
+	$(CODE ref S opAssign(ref S))  will be automatically generated.)
+
+        $(P A user-defined one can implement the equivalent semantics, but can
```
Comment (mihails-strasuns-sociomantic) [inline spec/struct.dd]: Indentation seems to be not in sync.

Comment (mihails-strasuns-sociomantic) [inline spec/struct.dd]: Also "but can one must" doesn't seem to be a well-formed English to me :)

**Comment by mihails-strasuns-sociomantic** [comment]
Thanks!

File: spec/struct.dd [hunk 2]
```diff
@@ -141,9 +141,9 @@ struct S;
 union U;
 ---
 
-	$(P The members are completely hidden to the user, and so the only operations
-	on those types are ones that do not require any knowledge of the contents
-	of those types. For example:)
+        $(P The members are completely hidden to the user, and so the only operations
```
Comment (andralex) [inline spec/struct.dd]: Are these hard tabs?

Comment (CyberShadow) [inline spec/struct.dd]: No. Trivial test: if you can select individual spaces (here, on this webpage even), then they're not hard tabs.

File: spec/struct.dd [hunk 3]
```diff
@@ -512,7 +512,7 @@ $(SECTION3 $(LEGACY_LNAME2 AssignOverload, assign-overload, Identity Assignment
         $(P where $(CODE opAssign) is a member function of S:)
 
         ---
-        ref S opAssign(S s)
+        ref S opAssign(ref S s)
```
Comment (9rnsr) [inline spec/struct.dd]: This is incorrect change. The assigned lvalue needs to be copied before the assignment.

Comment (ZombineDev) [inline spec/struct.dd]: @9rnsr I think you meant assignment from **rvalue**. In that case, is `ref S opAssign()(auto ref S s)` a more correct representation?

**Pull request merged**